tiledb/sm/query/query.cc

```diff
@@ -1214,8 +1219,15 @@ Status Query::init() {
       return logger_->status(Status_QueryError(errmsg.str()));
     }
 
+    // Check the buffer names.
     RETURN_NOT_OK(check_buffer_names());
 
+    // Set the timestamp to use for new fragments if the query type is write,
+    // modify exclusive, update or delete (any type but read).
+    if (type_ != QueryType::READ) {
```

Comment:
I'd suggest not to add more login in `init()` functions that are meant to be eliminated eventually. Is there a reason we can't initialize `current_fragment_timestamp_ `in `Query` constructor?

Reply:
We want the timestamp to be set as the time the Query is submitted/initialized, not first instantiated.